dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/filters/ApiAuthorizationFilter.java

```diff
@@ -0,0 +1,77 @@
+package edu.harvard.iq.dataverse.api.filters;
+
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.GuestUser;
+
+import javax.inject.Inject;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class ApiAuthorizationFilter implements Filter {
```

Comment:
Maybe some Javadocs explaining context for different people in the future. Like why do we need to check both session and token.